packages/typespec-azure-core/test/types/models.test.ts

```diff
@@ -0,0 +1,26 @@
+import { getHttpPart, isOrExtendsHttpFile } from "@typespec/http";
+import { ok, strictEqual } from "assert";
+import { it } from "vitest";
+import { getOperations } from "../test-host.js";
+
+it("FileWithRequiredMetadata could be recognized by @typespec/http", async () => {
+  const [operations, _, runner] = await getOperations(
+    `
+      model TestModel {
+        file: HttpPart<FileWithRequiredMetadata>;
+      };
+
+      @post op TestOperation(@header contentType: "multipart/form-date", @multipartBody body: TestModel): void;
+      `
+  );
+
+  ok(operations.length === 1);
+  ok(operations[0].parameters.body);
+  strictEqual(operations[0].parameters.body.bodyKind, "multipart");
+  strictEqual(operations[0].parameters.body.type.kind, "Model");
+  const fileHttpPart = operations[0].parameters.body.type.properties.get("file");
+  ok(fileHttpPart);
+  const file = getHttpPart(runner.program, fileHttpPart.type);
+  ok(file !== undefined);
+  ok(isOrExtendsHttpFile(runner.program, file.type));
```

Comment:
can you check the optionality of the `contentType` and `filename` properties? They should be required

Reply:
I added the check in TCGC test: https://github.com/Azure/typespec-azure/pull/1458/files#diff-5d0dbc69414fb42ee89e9e13679560f649d290268dca89d259d5567f0174a7ab